### ISSUE TITLE
error: Incompatible shapes: [600,27] vs. [50,27]

### ISSUE THREAD

**Issue by payman-kh** [fchollet/keras#4269]
Hello Githubers,

I am trying to design a CNN that has 12 parallel Convolutional2D layers which are later merged.
The input for each of these convolutional layers is a (1, 100, 100) image and I am setting the batch size to 50.  

here is how I defined each of convolutional layers and the fitting:

`

     kernel_size = 5
     filter_nr1 = 10   
     conv1 = Sequential()
     conv1.add(Convolution2D(filter_nr1,kernel_size,kernel_size, border_mode='same', dim_ordering = 'th', input_shape = (1,100,100), activation = 'relu'))
     conv1.add(MaxPooling2D(pool_size = (2,2), strides = (2,2), border_mode = 'same', dim_ordering = 'th'))
     conv1.add(Flatten())
    # and 11 other layers like this which are merged in the model

     model = create_model()  
     model.compile(loss='mean_squared_error', optimizer='SGD', metrics=['accuracy'])
    model.fit([x_train_0, x_train_1, x_train_2, x_train_3, x_train_4, x_train_5, x_train_6, x_train_7, x_train_8, x_train_9,x_train_10,x_train_11], y_train,  validation_data=([x_test_0, x_test_1, x_test_2, x_test_3, x_test_4, x_test_5, x_test_6, x_test_7, x_test_8, x_test_9, x_test_10, x_test_11], y_test), nb_epoch=4, batch_size=50)

`

sizes of the x_test and x_train sets are (500,100,100)
sizes of the y_test and y_train sets are (500,27)

But I get the error while fitting:
Incompatible shapes: [600,27] vs. [50,27]

I tried changing the input_shape to (50,1,100,100) but that also throws the error
Exception: Input 0 is incompatible with convolutian2D_1: expected ndim=4, found ndim=5

I am still havent learnt keras well and can not find a comprehensive reference. Can someone help me please?

thanks in advanced

**Comment by anewlearner** [comment]
You used `[x_train_0, x_train_1, x_train_2, x_train_3, x_train_4, x_train_5, x_train_6, x_train_7, x_train_8, x_train_9,x_train_10,x_train_11]` in the code. What do you mean by `sizes of the x_test and x_train sets are (500,100,100)`.
If the input shape of each branch is (1, 100, 100).  The shape of `x_train_0` should be (number of samples, 1, 100, 100). So as x_train_* and x_test_*.
